.github/workflows/push-image.yml

```diff
@@ -60,82 +113,113 @@ jobs:
+      if: ${{ needs.preparation.outputs.push_to_gcr == 'true' }}
+      with:
+        username: _json_key
+        password: ${{ secrets.GCR_PUSH_BOT_JSON_KEY }}
+        registry: gcr.io
 
-    - name: Push ${{ matrix.oci_image }} Image to DockerHub
+    - name: Push to DockerHub
       id: push
       env:
-        DOCKERHUB_USERNAME: ${{ secrets.PAKETO_BUILDPACKS_DOCKERHUB_USERNAME }}
-        DOCKERHUB_PASSWORD: ${{ secrets.PAKETO_BUILDPACKS_DOCKERHUB_PASSWORD }}
         DOCKERHUB_ORG: "${{ needs.preparation.outputs.DOCKERHUB_ORG }}"
-        GCR_USERNAME: _json_key
-        GCR_PASSWORD: ${{ secrets.GCR_PUSH_BOT_JSON_KEY }}
         GCR_PROJECT: "${{ github.repository_owner }}"
       run: |
-
-        if [ "${{ needs.preparation.outputs.push_to_dockerhub }}" == "true" ]; then
-          echo "${DOCKERHUB_PASSWORD}" | sudo skopeo login --username "${DOCKERHUB_USERNAME}" --password-stdin index.docker.io
-          sudo skopeo copy "oci-archive:${GITHUB_WORKSPACE}/${{ matrix.oci_image }}.oci" "docker://${DOCKERHUB_ORG}/${{ matrix.oci_image }}-${{ steps.registry-repo.outputs.name }}:${{ steps.event.outputs.tag }}"
-          sudo skopeo copy "oci-archive:${GITHUB_WORKSPACE}/${{ matrix.oci_image }}.oci" "docker://${DOCKERHUB_ORG}/${{ matrix.oci_image }}-${{ steps.registry-repo.outputs.name }}:latest"
+        # Ensure other scripts can access the .bin directory to install their own
+        # tools after we install them as whatever user we are.
+        mkdir -p ./.bin/
+        chmod 777 ./.bin/
+
+        ./scripts/publish.sh \
+          --build-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.build_image }}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+          --build-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.build_image }}-${{ needs.preparation.outputs.repo_name }}:latest" \
+          --run-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.run_image }}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+          --run-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.run_image }}-${{ needs.preparation.outputs.repo_name }}:latest" \
+          --build-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.build_image }}.oci"  \
+          --run-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.run_image }}.oci"
+
+        if [ "${{ needs.preparation.outputs.push_to_gcr }}" = "true" ]; then
+          for imageType in ${{ matrix.stack.build_image }} ${{ matrix.stack.run_image }}; do
+            echo "FROM docker.io/${DOCKERHUB_ORG}/${imageType}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" | \
+            docker buildx build -f - . \
+              --tag "gcr.io/${GCR_PROJECT}/${imageType}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+              --tag "gcr.io/${GCR_PROJECT}/${imageType}-${{ needs.preparation.outputs.repo_name }}:latest" \
+              --platform linux/amd64,linux/arm64 \
```

Comment:
As discussed explore how we can avoid hard coding, and test one repo where pushed to gcr are done.

Reply:
Added it but not tested pushing to GCR.